[ZEPPELIN- 1298] Log instead of throwing trace for ping messages

khalidhuseynov opened the pull request:
### What is this PR for?
When non authenticated user in non-anonymous mode tries to send any websocket api message it results in throwing a traceback. However on PING message we don't need to show the whole traceback and we can just log in order not to pollute the logs file.


### What type of PR is it?
Hot Fix

### Todos
* [x] - log and return on PING

### What is the Jira issue?
[Zeppelin-1298](https://issues.apache.org/jira/browse/ZEPPELIN-1298)

### How should this be tested?
Follow the steps in issue and shouldn't get repeating traceback.

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

Leemoonsoo commented:
Thanks @khalidhuseynov for the fix. I tried this patch and it does not print stacktrace on PING message anymore when user not logged in.

However, it still prints stacktrace when user browse zeppelin gui. And i think it's also the case we don't need to print stacktrace.
Could you handle this case, too?

And IMO, i think it's okay to adjust log message level to DEBUG for both case because user browse Zeppelin gui without valid ticket is fairly normal behavior, when they're not logged in.

khalidhuseynov commented:
@Leemoonsoo thanks for review. i addressed your comments by changing all the invalid cases to log in debug mode instead of throwing exception. let me know if there's something else left.

felixcheung commented on zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:
```diff
@@ -131,8 +131,12 @@ public void onMessage(NotebookSocket conn, String msg) {
-        throw new Exception("Invalid ticket " + messagereceived.ticket + " != " + ticket);
+      if (ticket != null && !ticket.equals(messagereceived.ticket)){
+        /* not to pollute logs, log in debug instead of exception */
+        LOG.debug("{} message: invalid ticket {} != {}", messagereceived.op,
```
LOG.warning instead?

Leemoonsoo replied:
When messagereceived.ticket is null or empty string, which is the case user not logged in and browse Zeppelin GUI, i think we can minimize the log in this case (log.debug?).

When messagereceived.ticket is not empty, but not equals to ticket, that means something wrong, and in this it's better print some logs.

khalidhuseynov replied:
@felixcheung @Leemoonsoo addressed in 3fa5403058b2d258180ecf23e8e7e1c82169dc84

bzz commented:
`LOG.debug` sounds like a great idea to me to avoid filling up the logs in default configuration.

LGTM

minahlee commented:
@khalidhuseynov Could you resolve conflicts?

khalidhuseynov commented:
@minahlee done

Leemoonsoo commented:
LGTM

Leemoonsoo commented:
Merge into master and branch-0.6 if there're no further discussions.
